### PR TITLE
Clean up property iterator when serializing property stores

### PR DESCRIPTION
**Pull request by pbardea**
Property iterators may be in arbitrary positions when serializing. Since they are not persisted, when the graph is deserialized the `propertyIterator` will be re-created. Therefore so the property stores looks as they so when serialized the iterators should be reset upon serialization.

### CODE REVIEW

File: src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java [hunk 1]
```diff
@@ -15,6 +16,13 @@
 
     protected PropertyIterator propertyIterator = new PropertyIterator();
 
+    @Override
+    public void serializeAll(String outputDirectoryPath) throws IOException, InterruptedException {
+        // Clean up property iterator when serializing because it is not persisted, so will not
+        // be deserialized.
+        propertyIterator.reset(null, 0, 0);
```
Comment (queryproc) [inline src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java]: specify in /* */ what that null is for instance /* data byte array */, 0 /* currentIndex */, 0 /* endIndex? */

File: src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java [hunk 2]
```diff
@@ -15,6 +16,13 @@
 
     protected PropertyIterator propertyIterator = new PropertyIterator();
 
+    @Override
+    public void serializeAll(String outputDirectoryPath) throws IOException, InterruptedException {
+        // Clean up property iterator when serializing because it is not persisted, so will not
```
Comment (queryproc) [inline src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java]: it does not need to be persisted. remove what is after.

Comment (pbardea) [reply src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java]: done

File: src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java (hunk 1 above)
Comment (pbardea) [reply src/main/java/ca/waterloo/dsg/graphflow/graph/PropertyStore.java]: done

**Pull request merged**